assets/pages/oneoff-contributions/components/oneoffContributionsPayment.jsx

```diff
@@ -51,33 +54,56 @@ function getStatusMessage(isFormEmpty: boolean, error: ?string): Node {
 function formValidation(
   isFormEmpty: boolean,
   error: ?string => void,
-): Function => void {
+): Function {
 
-  return (callback: Function) => {
+  return (): boolean => {
 
     if (!isFormEmpty) {
-      error(null);
-      callback();
-    } else {
-      error('Please fill in all the fields above.');
+      if (error) {
+        error(null);
+      }
+      return true;
     }
 
+    if (error) {
+      error('Please fill in all the fields above.');
+    }
+    return false;
   };
 
 }
 
 
 // ----- Component ----- //
 
-function OneoffContributionsPayment(props: PropTypes) {
+
+/*
+ * WARNING: we are using the React context here to be able to pass the getState function
+ * to the postCheckout function. PostCheckout requires this function to access to the
+ * most recent user.
+ * More information here:https://reactjs.org/docs/context.html
+ * You should not use context for other purposes. Please use redux.
+ */
```

Comment:
"If you want your application to be stable, don’t use context. It is an experimental API and it is likely to break in future releases of React." - if we're having to resort to this, is it possible that we need to rethink the way we've built things...?

Reply:
Yes, it wasn't my first option. I tried three different approaches before:

* Passing a copy of the state to the `postCheckout` function. It didn't work because Stripe was set up (`Stripe.configure`) with the first copy and it was not able to get the latest changes of the State.
* Creating a function called `getUserFromState` and pass it to `Stripe.configure` function. this option has the same flaw than the previous one.
* Set up Stripe every time the StripePopUp button was rendered. This option was functionally correct. Nevertheless, it generated an overhead of requests to Stripe. It generated three requests to Stripe every time the state change, i.e. everytime the user made a keystroke.

Since any of the approaches mentioned before didn't work I decided to go with the `context`  "solution". Nevertheless, it is worth to analyse why there aren't other options. The fact that we need the state in `stripePopUpButton` probably means that we need to tie that component to the redux state, but that clashes with our internal rule of shared components should not be smart components. Probably we need to have a further discussion about if that rule makes sense or not. Personally, I like that rule, makes the shared components less tangled.